assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -1012,6 +1013,49 @@ export default Component.extend({
     next(this.reStickScrollIfNeeded.bind(this));
   },
 
+  @discourseComputed("chatChannel.status")
+  channelStatusMessage(channelStatus) {
+    if (channelStatus === CHANNEL_STATUSES.open) {
+      return null;
+    }
+
+    switch (channelStatus) {
+      case CHANNEL_STATUSES.closed:
+        if (this.currentUser.staff) {
+          return I18n.t("chat.channel_status.closed_staff_header");
+        } else {
+          return I18n.t("chat.channel_status.closed_header");
+        }
+        break;
+      case CHANNEL_STATUSES.readOnly:
+        return I18n.t("chat.channel_status.read_only_header");
+        break;
+      case CHANNEL_STATUSES.archived:
+        return I18n.t("chat.channel_status.archived_header");
+        break;
+    }
+  },
+
+  @discourseComputed("chatChannel.status")
+  channelStatusIcon(channelStatus) {
+    if (channelStatus === CHANNEL_STATUSES.open) {
+      return null;
+    }
+
+    switch (channelStatus) {
+      case CHANNEL_STATUSES.closed:
+        return "lock";
+        break;
+      case CHANNEL_STATUSES.readOnly:
+        // FIXME (martin): Use a pencil-slash icon here, we don't have one in FA5
+        return "far-eye";
+        break;
+      case CHANNEL_STATUSES.archived:
+        return "folder";
+        break;
+    }
+  },
+
```

Comment:
I'm not sure if it's worth it to do this, but the `chat-live-pane` is one of those files that's just getting bigger, and bigger, and bigger...

I'm wondering if it's worth it to extract a `channel-status` component to isolate this logic.

Reply:
Great idea! I have moved it now :)